source/common/tcp_proxy/upstream.cc

```diff
@@ -77,12 +77,57 @@ TcpUpstream::onDownstreamEvent(Network::ConnectionEvent event) {
 
 HttpUpstream::HttpUpstream(Tcp::ConnectionPool::UpstreamCallbacks& callbacks,
                            const TunnelingConfigHelper& config,
-                           StreamInfo::StreamInfo& downstream_info)
+                           StreamInfo::StreamInfo& downstream_info, Http::CodecType type)
```

Comment:
`Http2Upstream` that is being removed was handling H3 as well. Please see [here](https://github.com/envoyproxy/envoy/blob/main/source/common/tcp_proxy/upstream.cc#L250-L252) that we are creating `Http2Upstream` for H3 type as well.

Before creating stream, type specific pool is created where [protocol] (https://github.com/envoyproxy/envoy/blob/main/source/common/tcp_proxy/upstream.cc#L235 ) is passed to communicate the type so that accordingly H3 pool gets initialized if type is H3.

Reply:
Missed that, thank you for the pointers!